### PR TITLE
Update to officially assigned port number and mdns service names for Matter protocol

### PR DESCRIPTION
**Pull request by aronrosenberg**
#### Problem
Update the code to use the new officially assigned port number and mdns service names from IANA.

 #### Summary of Changes
Change all instances of the old port number 11097 to 5540 in the code, documentation, examples and test scripts
Change all instances of mdns service name _chip to _matter
Change all instances of mdns service name _chipc to _matterc
Change all instances of mdns service name _chipd to _matterd

Fixes https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/1777

#### Testing
* Manually tested by compiling and checking advertisements and listening ports
* This is a breaking change and requires all sides to update as the port and service names have changed

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — CHANGES_REQUESTED** [review]
Please update the PR description to follow the standard template, which includes testing details, etc.

**Comment by aronrosenberg** [comment]
This PR is in Draft mode until IANA formally issues the allocation. Once that confirmation is received, I will mark as ready for commiting.

**Comment by aronrosenberg** [comment]
The IANA submission is still in progress.

**Comment by aronrosenberg** [comment]
A clarification question was asked by the IESG expert which means their review is still in progress.

File: src/darwin/Framework/CHIPTests/CHIPClustersTests.m [hunk 1]
```diff
@@ -32,8 +32,8 @@
 const uint64_t kDeviceId = 1;
 const uint16_t kDiscriminator = 3840;
 const uint32_t kSetupPINCode = 20202021;
-const uint16_t kRemotePort = 11097;
-const uint16_t kLocalPort = 11098;
+const uint16_t kRemotePort = 5540;
+const uint16_t kLocalPort = 5541;
```
Comment (turon) [inline src/darwin/Framework/CHIPTests/CHIPClustersTests.m:36]: Should this be an ephemeral port?

**Review by turon — APPROVED** [review]

**Comment by aronrosenberg** [comment]
Feedback from IANA still in progress.

**Pull request closed without merging**